src/esm_runscripts/filedicts.py

```diff
@@ -45,30 +47,93 @@ def __init__(self, full_config, attrs_address):
-            "exp_tree": pathlib.Path(full_config["general"]["exp_dir"]),
-            "run_tree": pathlib.Path(full_config["general"]["thisrun_dir"]),
+            "work": pathlib.Path(full_config[component]["thisrun_work_dir"]),
+            "pool": pathlib.Path(self["path_in_pool"]),
+#            "exp_tree": pathlib.Path(full_config[component]["exp_dir"]), # This name is incorrect and depends on the type of file (to be resolved somewhere else before feeding it here)
+#            "run_tree": pathlib.Path(full_config[component]["thisrun_dir"]), # This name is incorrect and depends on the type of file (to be resolved somewhere else before feeding it here)
+
+        }
+        self.names = {
+            "work": pathlib.Path(self["name_in_work"]),
+            "pool": pathlib.Path(self["name_in_pool"]),
         }
         # Allow dot access:
         self.work = self.locations["work"]
         self.pool = self.locations["pool"]
-        self.exp_tree = self.locations["exp_tree"]
-        self.run_tree = self.locations["run_tree"]
+#        self.exp_tree = self.locations["exp_tree"] # TODO: uncomment when lines above are fixed
+#        self.run_tree = self.locations["run_tree"] # TODO: uncomment when lines above are fixed
 
-    def cp(self) -> None:
-        pass
+        # Verbose set to true by default, for now at least
+        self.verbose = full_config.get("general", {}).get("verbose", True)
+
+    def _cp(self, source, target) -> None:
+        """
+        Copies the source file or folder to the target path. It changes the name of the
+        target if ``self["name_in_<target>"]`` differs from ``self["name_in_<source>"].
+
+        Parameters
+        ----------
+        source : str
+            String specifying one of the following options: ``"pool"``, ``"work"``,
+            ``"exp_tree"``, ``run_tree``
+        target : str
+            String specifying one of the following options: ``"pool"``, ``"work"``,
+            ``"exp_tree"``, ``run_tree``
+        """
+        # Build target and source paths
+        spath = self.locations[source].joinpath(self.names[source])
+        tpath = self.locations[target].joinpath(self.names[target])
+
+        # Checks
+        spath_type = self._check_source_and_target(spath, tpath)
 
-    def ln(self) -> None:
+        # Actual copy
+        try:
+            shutil.copy2(spath, tpath)
+        except Exception:
+            user_error("Filedict Error", f"Unable to copy {spath} to {tpath}")
+
+    def _ln(self) -> None:
         pass
 
-    def mv(self) -> None:
+    def _mv(self) -> None:
         pass
-        
-        
-def copy_files(config):
-    """Copies files"""
-    # PG: No. We do not want this kind of general function. This is just to
-    # demonstrate how the test would work
-    return config
+
+    def _path_type(self, path):
+        if path.is_file():
+            return "file"
+        elif path.is_dir:
+            return "dir"
+        elif path.is_link:
```

Comment:
This should be `islink()`. Ie. `os.path.islink()`

Reply:
fixed